### DISCUSSION TITLE
The reference of `formState.errors` state remains constant which doesnot trigger React's useEffect hook

### DISCUSSION THREAD

**Discussion by Avas77** [react-hook-form/react-hook-form#8449] [Q&A]
@bluebill1049 

Hello Bill,
I have been trying to upgrade the use of RHF from v5 to v7. In v5, the reference of errors object in formState changed when form validation is triggered. Based on the change in reference, I had been scrolling to the specific field manually. After the upgrade it is no longer possible to do so as the refernce stays the same the useEffect hook is not triggered.

I have created a method named handleErrors in my parent component like this:
`  const handleErrors = React.useCallback(
      (errors: FieldErrors<FieldValues>) => {
        Object.keys(errors).every((fieldName) => {
          scrollIntoViewAndFocus(fieldName);
          return false;
        });
      },
      []
    );`

I have passed this method as a prop called onErrors to my child component and invoked it in the useEffect hook like this:
` React.useEffect(() => {
    onErrors?.(errors);
  }, [errors, onErrors]);`

This is not working as the reference of the error stays the same. I tried putting the whole formState as a dependecy in useEffect hook, 
the scrolling works on invalid fields but a different issue arises with this approach, whenever the field becomes valid it immediately scrolls to the next invalid field, which is not the behaviour I want in my application

**Comment by bluebill1049** [comment]
with v7 formState is a batch update, you will need to put formState as dep, in terms of the individual state as useEffect dep that would likely to made into v8, since react 18 has a batch update now.
